test/SolidityNameAndTypeResolution.cpp

```diff
@@ -321,6 +323,23 @@ BOOST_AUTO_TEST_CASE(comparison_bitop_precedence)
 	BOOST_CHECK_NO_THROW(parseTextAndResolveNames(text));
 }
 
+BOOST_AUTO_TEST_CASE(function_canonical_signature)
```

Comment:
Please don't forget to add a test with "uint", "hash" (which should be uint256 and hash256 in the signature) and "address" (which should stay as it is).